File: weirentao/src/main/java/com/example/demo/controller/UserController.java
package com.example.demo.controller;

import com.example.demo.entity.User;
import com.example.demo.mapper.UserMapper;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;
import springfox.documentation.swagger2.annotations.EnableSwagger2;

import java.util.List;

@Api(tags="user")
@RestController
@EnableSwagger2
public class UserController {
    @Autowired
    private UserMapper userMapper;
    @ApiOperation(value ="用户添加")
    @GetMapping("/addUser")
     public int AddUser(User user){
        if(user!=null){
            if(user.getId()!=0) {
                //是否可以添加此主键
            }
        }
         return userMapper.insertUser(user);
     }
    @ApiOperation(value ="更新信息")
    @PostMapping("/addUser")
    public int updateUser(User user){
        return userMapper.updateUser(user);
    }
    @ApiOperation(value ="删除信息")
    @GetMapping("/deleteUser")
    public int deleteUser(int id){
        return userMapper.deleteUser(id);
    }
    @ApiOperation(value ="查询信息")
    @GetMapping("/selectAllUsers")
    public List<User> selectAllUsers(){
        return userMapper.selectAllUsers();
    }
}

File: caiguicheng/week01/day03/第三天上午 使用注解的数据库使用/demo03/controller/UserController.kt
package com.example.demo03.controller

import com.example.demo03.entity.User
import com.example.demo03.mapper.UserMapper
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController
import springfox.documentation.swagger2.annotations.EnableSwagger2

/**
 *
 */
@Api(tags = ["user"])
@RestController
@EnableSwagger2
class UserController {

    @Autowired
    lateinit var userMapper: UserMapper

    @ApiOperation("用户添加")
    @GetMapping("/addUser")
    fun addUser(user: User): Int {
        if (user.id!=0) {
            return userMapper.insertUser(user)
        }
        return 0
    }

    @ApiOperation("删除用户")
    @DeleteMapping("/deleteUser")
    fun deleteUser(id: Int): Int {
        return userMapper.deleteUser(id)
    }

    @ApiOperation("修改用户信息")
    @PostMapping("/updateUser")
    fun updateUser(user: User): Int {
        return userMapper.updateUser(user)
    }

    /**
     * 根据条件获得用户
     * @return 符合条件的用户列表
     */
    @ApiOperation("查找用户信息")
    @PostMapping("/selectUser")
    fun selectUser(user: User): List<User> {
        println("time")
        println(user.createTime)
        return userMapper.selectUser(user)
    }
}